src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -244,19 +249,35 @@ public static long countBulkFileLoadModelsFromPath(
       ObjectMapper objectMapper,
       IngestRequestModel ingestRequest,
       String cloudEncapsulationId,
-      List<String> fileRefColumnNames,
+      List<Column> fileRefColumns,
       List<String> errors) {
     return IngestUtils.getJsonNodesStreamFromFile(
             cloudFileReader, objectMapper, ingestRequest, cloudEncapsulationId, errors)
         .flatMap(
             node ->
-                fileRefColumnNames.stream()
-                    .map(node::get)
-                    .filter(n -> n != null && n.isObject())
+                fileRefColumns.stream()
+                    .map(c -> node.get(c.getName()))
```

Comment:
Could write this as two separate `map`s
```suggestion
                    .map(Column::getName)
                    .map(node::get)
```